spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -280,8 +282,8 @@ public boolean hasNext() {
 				}
 				catch (IOException e) {
 					try {
-						bufferedReader.close();
 						this.done = true;
+						bufferedReader.close();
```

Comment:
Why did you swap these lines?

Reply:
The reason the same as in `close()` method below:
if `bufferedReader.close();` thows any exception, then `this.done = true;` will not run and we end up in undeterministic state.